### ISSUE TITLE
Error when omitting optional inputs within a model local function

### ISSUE THREAD

**Issue by jantonguirao** [microsoft/onnxruntime#10249]
**Describe the bug**
Using an operator with not provided optional inputs within a model local function results in an error `"InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid model. Node input '0x2464d70' is not a graph input, initializer, or output of a previous node."`

**Urgency**
This slows down the progress of the experiments regarding the ONNX preprocessing working group.

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 20.04.3 LTS
- ONNX Runtime installed from (source or binary): binary
- ONNX Runtime version: 1.10.0
- Python version: Python 3.8.10
- Visual Studio version (if applicable): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**To Reproduce**
This code works:
```
m = parser.parse_model('''
<
  ir_version: 8,
  opset_import: [ "" : 14, "local" : 1 ],
  producer_name: "test",
  producer_version: "1.0",
  model_version: 1,
  doc_string: "Test preprocessing model"
>
agraph (float[N] x) => (float[N] x_resized)
{
    k2f = Constant <value = float[1] {2.0}> ()
    x_resized = Resize (x, , k2f)
}
''')

checker.check_model(m)
# Save the model to a file
save(m, 'test_ok.onnx')

session = onnxruntime.InferenceSession('test_ok.onnx', None)
result = session.run([], {'x': np.array([1.0, 2.0, 3.0], dtype=np.float32)})
print(result)
```
producing
```
[array([1., 1., 2., 2., 3., 3.], dtype=float32)]
```

However, this code fails:
```
m2 = parser.parse_model('''
<
  ir_version: 8,
  opset_import: [ "" : 14, "local" : 1 ],
  producer_name: "test",
  producer_version: "1.0",
  model_version: 1,
  doc_string: "test"
>
agraph (float[N] x) => (float[N] x_resized)
{
    x_resized = local.resize_test (x)
}

<
  opset_import: [ "" : 14, "local" : 1 ],
  domain: "local",
  doc_string: "test"
>
resize_test (x) => (x_resized) {
    k2f = Constant <value = float[1] {2.0}> ()
    x_resized = Resize <mode = \"linear\"> (x, , k2f)
}
''')
checker.check_model(m2)
# Save the model to a file
save(m2, 'test_fail.onnx')

session = onnxruntime.InferenceSession('test_fail.onnx', None)
result = session.run([], {'x': np.array([1.0, 2.0, 3.0], dtype=np.float32)})
print(result)
```
resulting in the following error:
```
InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid model. Node input '0x2d3a660' is not a graph input, initializer, or output of a previous node.
```

**Expected behavior**
Both models should produce same results

**Screenshots**
N/A

**Additional context**
N/A

**Comment by tianleiwu** [comment]
onnxruntime support Node with optional inputs (The input name shall be empty string). In your case, there is input name '0x2d3a660' so the input is provided but not existed. Could you double check that the onnx model is expected with netron visualizer?

**Comment by jantonguirao** [comment]
As I mentioned earlier, the first model that doesn't use a local function works as expected. 
When I visualize the model with netron, I can see that the optional input is not provided. However, it is when I create the ORT InferenceSession that the issue occurs.
![image](https://user-images.githubusercontent.com/3891217/149022555-5a04280a-4ca3-4906-8512-b97418b3c08b.png)

**Comment by jantonguirao** [comment]
@tianleiwu 
Any updates on this issue?

**Comment by gramalingam** [comment]
Hi, I believe this is likely a bug in the function-inlining implementation in ORT. The code renaming inputs/outputs of a node should check if it is a missing optional input/output and skip renaming those. Eg., see: https://github.com/microsoft/onnxruntime/blob/3ae2bfaefe4344820d0a17055aa23f080530af34/onnxruntime/core/graph/function.cc#L672 and https://github.com/microsoft/onnxruntime/blob/3ae2bfaefe4344820d0a17055aa23f080530af34/onnxruntime/core/graph/function.cc#L624

**Comment by gramalingam** [comment]
I ran into the same problem (I think) in another PR, and I added a fix in that PR. See https://github.com/microsoft/onnxruntime/pull/10779 
Once the above PR is merged, it may be useful to see if that resolves this issue also.

**Comment by jantonguirao** [comment]
@gramalingam I can confirm that https://github.com/microsoft/onnxruntime/pull/10779 fixes this problem. Thank you.
